### DISCUSSION TITLE
MolFromSmiles return None even with sanitize=False

### DISCUSSION THREAD

**Discussion by LanceKnight** [rdkit/rdkit#4342] [Q&A]
Hello, as stated in the title, I still got a None. Can someone help me?
![image](https://user-images.githubusercontent.com/5760199/125685681-fbd7b063-adcc-417c-b934-6cb7349a6441.png)

rdkit version: 2020.09.1.0
python version :3.6.12

**Comment by ptosco** [comment]
@LanceKnight Sulfoximines do not display geometric isomerism, but they can bear a stereocenter at tetrahedral sulfur if the substituents are different, as in your case.
If you remove the _trans_ double bond specification from your SMILES string the molecule will build and the RDKit will correctly identify the tetrahedral stereocenter:
```
from rdkit import Chem
from rdkit.Chem.Draw import IPythonConsole

IPythonConsole.drawOptions.addAtomIndices = True

smi = r"S(=O)(=O)(N=S(=O)(c1ccc(cc1)C)C)c1c2nsnc2ccc1"
mol = Chem.MolFromSmiles(smi)
mol
```
![image](https://user-images.githubusercontent.com/5244385/125694005-f912cc73-75d3-4229-b38f-a6dd02b2bb83.png)
```
stereo_info = Chem.FindPotentialStereo(mol)
len(stereo_info)
1
stereo_info[0].centeredOn
4
stereo_info[0].type
rdkit.Chem.rdchem.StereoType.Atom_Tetrahedral
```

**Reply by LanceKnight** [reply]
I see. I'm actually a computer science student and not that familiar with chemistry. I'm using a dataset that contains lots of smiles looking like this. Is there a way to automatically correct it?

**Reply by greglandrum** [reply]
I think Paolo's answer wasn't necessarily clear.
The problem here is that the syntax `/=O` is not recognized by the RDKit SMILES parser and results in a syntax error. I don't believe that this is legal SMILES (at least not legal Daylight SMILES and I also don't think that syntax is supported by OpenSMILES).

If you can't correct the input so that it's legal SMILES, you should be able to "fix" this problem without losing information (as Paolo points out) by doing a string replacement and replacing all occurances of `/=` or `\=` with `=`

FWIW: if you look at the output when you try and parse that SMILES, you'll see that the code is reporting a syntax error in the SMILES. This would have told you directly what the problem is. Here's an example:
```
In [4]: m = Chem.MolFromSmiles('C/N=S(/=O)C')
[06:15:11] SMILES Parse Error: syntax error while parsing: C/N=S(/=O)C
[06:15:11] SMILES Parse Error: Failed parsing SMILES 'C/N=S(/=O)C' for input: 'C/N=S(/=O)C'
```

**Reply by LanceKnight** [reply]
Hi, Greg, Thanks for the answer. I will just replace them as you suggested :)

Btw, the error message is not showing in Jupyter even if I tried your method mentioned in [link](https://github.com/rdkit/rdkit/issues/834). The error message is shown as the screenshot in my question description at the top.

**Reply by greglandrum** [reply]
@LanceKnight are you using Windows?

**Reply by LanceKnight** [reply]
Yes. So the solution won't work for windows?

**Reply by greglandrum** [reply]
Unfortunately we still haven’t managed to get the logs to show up reliably on windows

**Reply by LanceKnight** [reply]
I see. Thank you! :)
